source/infrastructure/lib/api/model-schema/deploy-usecase-body.ts

```diff
@@ -58,6 +58,10 @@ export const deployUseCaseBodySchema: JsonSchema = {
             description: 'Deploy the CloudFront based UI for the use case',
             default: true
         },
+        ExistingCognitoUserPoolId: {
```

Comment:
Rather than having this parameter at the top level of the payload, I would perhaps suggest we create a `CognitoParams` subschema here containing `ExistingCognitoUserPoolId` and any other potential future params related to cognito. Or perhaps even a generic `AuthenticationParams` which itself contains provider specific auth params, which for now would just be `CognitoParams`. This would follow a similar pattern to what we use for KnowledeBaseParams and LlmParams.

Open to suggestions from your end.

Reply:
I agree, but I believe it would make sense to consider in the future releases, for the same of backward compatibility with the current used versions of GAAB. Introducing CognitoParams will mean that all currently deployed GAAB instances should have a way to migrate, i.e. set CognitoParams from ExistingCognitoUserPoolId (currently in use). 

If you agree, we can work on it in the next iterations, as an enhancement to this, and we can possibly consider extending with more user management functionality as well.

Reply:
Well since existing versions do not have the ability to set a custom user pool, adding `CognitoParams` as a non-required param would not break compatibility as far as I can tell. To continue using the default (created for you) pool, the front end would just not send that object in the request. On the other hand, if we add this and then change the positioning of it later, that certainly will break API compatibility.